tests/kernel/gen_isr_table/src/main.c

```diff
@@ -98,6 +98,15 @@ void isr4(void *param)
 	trigger_check[ISR4_OFFSET]++;
 }
 
+/* Need to turn optimization off. Otherwise compiler may generate incorrect
+ * code, not knowing that trigger_irq() affects the value of trigger_check,
+ * even if declared volatile.
+ *
+ * A memory barrier does not help, we need an 'instruction barrier' but GCC
+ * doesn't support this; we need to tell the compiler not to reorder memory
+ * accesses to trigger_check around calls to trigger_irq.
+ */
+__attribute__((optimize("-O0")))
```

Comment:
This idea of using the optimize to somehow give an instruction barrier makes me uneasy.  Why not just add a __ISB() for ARM architectures.   I am sure other arches have similar solutions.

Reply:
i don't want to get too sidetracked by this -- I needed to get rid of the global -O0 here because it messes up the padding size for the generated gperf code in the linker script.

I moved it to the function instead. if we want to further iterate on it, please file a bug.